server/src/main/java/org/apache/cassandra/sidecar/config/yaml/ServiceConfigurationImpl.java

```diff
@@ -149,6 +152,9 @@ public class ServiceConfigurationImpl implements ServiceConfiguration
     @JsonProperty(value = COORDINATION)
     protected final CoordinationConfiguration coordinationConfiguration;
 
+    @JsonProperty(value = DNS_RESOLVER_PROPERTY)
+    protected final DnsResolverEnum dnsResolver;
```

Comment:
The type here should be `DnsResolver`
```suggestion
    protected final DnsResolver dnsResolver;
```